core/wiki/macros/CSS.tid

```diff
@@ -1,39 +1,76 @@
 title: $:/core/macros/CSS
 tags: $:/tags/Macro
 
-\procedure actions-compile-palette(inputTitle,outputTitle)
-<!-- Redefine colour function to use the input palette -->
-\function colour(name)
-	[<inputTitle>getindex<name>] :else[[$:/config/DefaultColourMappings/]addsuffix<name>get[text]]
-\end colour
-<!-- Process modern filter-based palettes -->
-<%if [<inputTitle>get[palette-type]match[modern]] %>
-	<$list filter="[<inputTitle>indexes[]sort[]]" variable="colour-name">
-		<$let colour-filter={{{ [<inputTitle>getindex<colour-name>] }}}>
+\procedure actions-compile-palette-filtered(tempPalette,outputTitle)
+<!-- Make the colour function use the input palette -->
```

Comment:
Is it true, that procedures are a little bit more performant it they are outside the definition?

Reply:
Perhaps marginally because there is the possibility of caching the parse tree. But in this case this is an action that is performed once, and not something that is repeated with each refresh cycle, and so performance is not the primary concern.